systemd/systemd.go

```diff
@@ -931,57 +937,123 @@ func (s *systemd) IsActive(serviceName string) (bool, error) {
 	return false, err
 }
 
-func (s *systemd) Stop(serviceNames []string, timeout time.Duration) error {
+func (s *systemd) Stop(serviceNames []string) error {
 	if s.mode == GlobalUserMode {
 		panic("cannot call stop with GlobalUserMode")
 	}
-	if _, err := s.systemctl(append([]string{"stop"}, serviceNames...)...); err != nil {
-		return err
-	}
 
-	// and now wait for it to actually stop
-	giveup := time.NewTimer(timeout)
-	notify := time.NewTicker(stopNotifyDelay)
-	defer notify.Stop()
-	check := time.NewTicker(stopCheckDelay)
-	defer check.Stop()
-
-	firstCheck := true
-loop:
-	for {
-		select {
-		case <-giveup.C:
-			break loop
-		case <-check.C:
-			allStopped := true
+	// Start the real time progress tracking inside a go routine because the
+	// 'systemctl stop' request is blocking (which runs in the parent function).
+	// Note that although the status polling thread is separate, and could start
+	// before the 'systemctl stop' command, the poll ticker will not fire
+	// immediately, allowing the stop command to start first. The ordering is
+	// not assumed, but it helps make existing unit-test code work, because the
+	// systemctl mocking in some modules are implemented very simplistically, and
+	// assumes that the 'stop' argument comes before the 'show'.
+	errorRet := make(chan error)
+	exitSig := make(chan interface{})
```

Comment:
Nitpick^2, we typically use `quit` for such channel.

Reply:
Done.